src/MongoDB/WriteResult.c

```diff
@@ -136,8 +142,9 @@ PHP_METHOD(WriteResult, getUpsertedCount)
 		return;
 	}
 
-
-	RETURN_LONG(intern->write_result.nUpserted);
+	if (bson_iter_init_find(&iter, intern->reply, "nUpserted") && BSON_ITER_HOLDS_INT32(&iter)) {
+		RETURN_LONG(bson_iter_int32(&iter));
+	}
```

Comment:
I thought about it, we had other macros (e.g. `SCP()`) for copying over like fields, but wasn't sure if nested macros were kosher.